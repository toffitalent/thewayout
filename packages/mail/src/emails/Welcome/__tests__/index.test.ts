import { renderTemplate } from '@test';
import { Welcome } from '..';

describe('Welcome', () => {
  const context = {
    user: {
      firstName: 'Test',
    },
  };

  const email = new Welcome(context);

  test('renders correctly', async () => {
    expect(await email.render()).toMatchSnapshot();
    expect(renderTemplate('Welcome', 'html', context)).toMatchSnapshot();
    expect(renderTemplate('Welcome', 'subject', context)).toMatchSnapshot();
    expect(renderTemplate('Welcome', 'text', context)).toMatchSnapshot();
  });
});
