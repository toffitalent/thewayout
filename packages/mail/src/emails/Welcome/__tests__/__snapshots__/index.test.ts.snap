// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
{
  "html": "pug-test-stub",
  "subject": "pug-test-stub",
  "text": "pug-test-stub",
}
`;

exports[`Welcome renders correctly 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta name="viewport" content="width=device-width, initial-scale=1.0"/><meta name="x-apple-disable-message-reformatting"/><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="color-scheme" content="light dark"/><meta name="supported-color-schemes" content="light dark"/><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nunito:400,700&amp;display=swap"/><link rel="stylesheet" href="@app/styles/index.scss" data-inline="true"/><!--[if mso]><style type="text/css">.f-fallback {
  font-family: Arial, sans-serif;
}</style><![endif]--></head><body><span class="preheader">Welcome to The Way Out, Test! We're thrilled to have you on board!</span><table class="email__wrapper" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td align="center"><table class="email__content" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="email__masthead"><a class="f-fallback email__masthead__logo" href="http://__WEB__"><img src="@app/assets/logo.png" alt="The Way Out"/></a></td></tr><tr><td class="email__body" width="570" cellpadding="0" cellspacing="0"><table class="email__body__inner" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content"><div class="f-fallback"><h1>Welcome, Test</h1><p>Thanks for signing up for The Way Out. We're thrilled to have you on board!</p><p>To get started, be sure to create your startup profile so people can see what you're working on:</p><table class="body__action" align="center" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td align="center"><table width="100%" border="0" cellspacing="0" cellpadding="0" role="presentation"><tr><td align="center"><a class="f-fallback button button--primary" href="http://__WEB__" target="_blank">Create Profile</a></td></tr></table></td></tr></table><p>If you have any questions, feel free to
<a href="mailto:support@thewayouthelps.com">email our support team</a>. We're always happy to help :)</p><p>Thanks,<br/>The The Way Out Team</p><table class="body__sub" role="presentation"><tr><td><p class="f-fallback sub">If you&apos;re having trouble with the button above, copy and paste the URL below into your web browser.</p><p class="f-fallback sub"><a class="nocolor" href="http://__WEB__">http://__WEB__</a></p></td></tr></table></div></td></tr></table></td></tr><tr><td><table class="email__footer" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content" align="center"><p class="f-fallback sub align-center">&copy; 2024 The Way Out. All rights reserved.</p><p class="f-fallback sub align-center">The Way Out, Inc.<br/><a href="#">PO Box 70620</a><br/><a href="#">Milwaukee, WI 53207</a></p></td></tr></table></td></tr></table></td></tr></table></body></html>"
`;

exports[`Welcome renders correctly 3`] = `"Getting started with The Way Out"`;

exports[`Welcome renders correctly 4`] = `
"Welcome, Test!

Thanks for signing up for The Way Out. We're thrilled to have you on board!

To get the most out of your new account, be sure to finish setting up your account:

My Account: http://__WEB__

If you have any questions, feel free to email our support team (support@thewayouthelps.com). We're always happy to help :)

Thanks,
The The Way Out Team

--
Â© 2024 The Way Out. All rights reserved.

The Way Out, Inc.
PO Box 70620
Milwaukee, WI 53207"
`;
